lib/devfs/include/devfs/device.h

```diff
@@ -190,22 +219,158 @@ devtoname(struct device *dev)
 	return dev->name;
 }
 
+/*
+ * Opens the device specified by name.
+ * This may be used to check whether a device exists or not.
+ *
+ * @param name
+ * 	The name of the device
+ * @param mode
+ * 	The open mode(should be handled by the device driver if needed)
+ * @param devp
+ * 	Return parameter. The opened device will be reachable in devp after
+ * 	the function returns.
+ * @return
+ * 	The result of the device's open routine
+ */
+
 int device_open(const char *name, int mode, struct device **devp);
+
+/*
+ * Closes the device specified by dev.
+ * The function does not return an error if the target driver does not have
+ * a close routine.
+ *
+ * @param dev
+ * 	The device to be closed
+ * @return
+ * 	The result of the device's close routine
+ */
+
 int device_close(struct device *dev);
+
+/*
+ * Performs a scatter/gather read on the device pointed to by dev.
+ *
+ * @param dev
+ * 	The device to read from
+ * @param uio
+ * 	The structure containing the scatter-gather list
+ * @param ioflags
+ * 	The flags to be sent to the read operation
+ * @return
+ * 	The result of the read operation
+ */
+
 int device_read(struct device *dev, struct uio *uio, int ioflags);
+
+/*
+ * Performs a scatter/gather write on the device pointed to by dev.
+ *
+ * @param dev
+ * 	The device to write to
+ * @param uio
+ * 	The structure containing the scatter-gather list
+ * @param ioflags
+ * 	The flags to be sent to the write operation
+ * @return
+ * 	The result of the write operation
+ */
+
 int device_write(struct device *dev, struct uio *uio, int ioflags);
+
+/*
+ * Performs an I/O control request.
+ *
+ * @param dev
+ * 	The device to perform the request on
+ * @param cmd
+ * 	The device dependent command to be executed
+ * @param arg
+ * 	The user buffer sent as argument to the command. The buffer
+ * 	must be validated by the ioctl routine of the driver.
+ * @return
+ * 	The result of the I/O control request
+ */
 int device_ioctl(struct device *dev, unsigned long cmd, void *arg);
+
+/*
+ * Returns information about a device.
+ * The target of the query should be placed in the cookie field
+ * of the parameter.
+ *
+ * @param info
+ * 	Used as a return parameter. The device must be specified
+ * 	in the cookie field and on return the id, flags and name
+ * 	fields will be filled with the requested information.
+ * @return
+ * 	0 on success and ESRCH otherwise
+ */
+
 int device_info(struct devinfo *info);
 
 int bdev_read(struct device *dev, struct uio *uio, int ioflags);
```

Comment:
Please also comment this function.

Reply:
Same as with `read_partition_table`, I could not find any definition or reference to it in the repository. The only reference I could find for them is [here](https://github.com/cloudius-systems/osv/blob/17bb6ae35c31b79615913a4a0363dae0a151c6f6/fs/vfs/vfs_bdev.cc).